src/ledger-bridge-keyring.test.ts

```diff
@@ -0,0 +1,845 @@
+      expect(accounts[2]?.address, fakeAccounts[2]);
+      expect(accounts[3]?.address, fakeAccounts[3]);
+      expect(accounts[4]?.address, fakeAccounts[4]);
+    });
+  });
+
+  describe('getPreviousPage', function () {
+    it('should return the list of accounts for current page', async function () {
+      // manually advance 1 page
+      await keyring.getNextPage();
+      const accounts = await keyring.getPreviousPage();
+
+      assert.equal(accounts.length, keyring.perPage);
+      expect(accounts[0]?.address, fakeAccounts[0]);
+      expect(accounts[1]?.address, fakeAccounts[1]);
+      expect(accounts[2]?.address, fakeAccounts[2]);
+      expect(accounts[3]?.address, fakeAccounts[3]);
+      expect(accounts[4]?.address, fakeAccounts[4]);
+    });
+
+    it('should be able to go back to the previous page', async function () {
+      // manually advance 1 page
+      await keyring.getNextPage();
+      const accounts = await keyring.getPreviousPage();
+
+      assert.equal(accounts.length, keyring.perPage);
+      expect(accounts[0]?.address, fakeAccounts[0]);
+      expect(accounts[1]?.address, fakeAccounts[1]);
+      expect(accounts[2]?.address, fakeAccounts[2]);
+      expect(accounts[3]?.address, fakeAccounts[3]);
+      expect(accounts[4]?.address, fakeAccounts[4]);
+    });
+  });
+
+  describe('getAccounts', function () {
+    const accountIndex = 5;
+    let accounts: string[] = [];
+    beforeEach(async function () {
+      keyring.setAccountToUnlock(accountIndex);
+      await keyring.addAccounts();
+      accounts = await keyring.getAccounts();
+    });
+
+    it('returns an array of accounts', function () {
+      assert.equal(Array.isArray(accounts), true);
+      assert.equal(accounts.length, 1);
+    });
+
+    it('returns the expected', function () {
+      const expectedAccount = fakeAccounts[accountIndex];
+      assert.equal(accounts[0], expectedAccount);
+    });
+  });
+
+  describe('exportAccount', function () {
+    it('should throw an error because it is not supported', function () {
+      expect(() => {
+        keyring.exportAccount();
+      }).to.throw('Not supported on this device');
+    });
+  });
+
+  describe('forgetDevice', function () {
+    it('should clear the content of the keyring', async function () {
+      // Add an account
+      keyring.setAccountToUnlock(0);
+      await keyring.addAccounts();
+
+      // Wipe the keyring
+      keyring.forgetDevice();
+
+      const accounts = await keyring.getAccounts();
+
+      assert.equal(keyring.isUnlocked(), false);
+      assert.equal(accounts.length, 0);
+    });
+  });
+
+  describe('signTransaction', function () {
+    describe('using old versions of ethereumjs/tx', function () {
+      it('should pass serialized transaction to ledger and return signed tx', async function () {
+        await basicSetupToUnlockOneAccount();
+        sandbox.on(
+          keyring?.iframe?.contentWindow as Window,
```

Comment:
As we throw an error if the iframe has not been initialized yet (see https://github.com/MetaMask/eth-ledger-bridge-keyring/pull/174#discussion_r1134033556), I added a function that does this check and throws an error if these objects are not what we expect to be. I also added a call to `simulateIFrameLoad` in the `beforeEach` so that we
manually call `keyring.iframe.onload`